### DISCUSSION TITLE
Measuring Devops Success - literature survey

### DISCUSSION THREAD

**Discussion by tracymiranda** [cdfoundation/end-user-council#4] [Ideas]
Measuring DevOps or Software Delivery Performance is well documented in [Accelerate](https://nicolefv.com/book) by Dr. Forsgren et al in Chapter 2. 

In addition to Accelerate, what other writings, articles, blog posts have you found helpful on this topic? Please share them here with a short description or key takeaways summarized. 👇👇👇

**Comment by roxannejoncas** [comment]
This cdCon 2020 talk "Measuring DevOps" by Dina Graves Portman (Google) is one our our most viewed videos. She also show an example using GitHub and Tekton. Developers have reached out to say it was useful.
Talk: https://youtu.be/rbenCqQCa-4

**Comment by amfred** [comment]
Craig and I recently gave a talk at the DevOps Enterprise Summit about DevOps Metrics That Matter, where we go through all of the metrics we gather automatically, why we think they're valuable, and what behavior we're trying to encourage.  I thought you might like that as input to your conversation.  See the Metrics in Detail for the actual metrics and the goals of measuring next.

https://github.com/devopsenterprise/2020-London-Virtual/blob/master/Day%202/Breakouts/Ann%20Marie%20Fred%20-%20Fred_Cook_June24_1355.pdf

Just this week we were looking to port our most valuable metrics from our Squad Metrics Dashboard into a new dashboard.  From that discussion, we decided that these were the DevOps metrics that are giving us the most mileage right now.
Availability vs. SLA/SLO attainment
SAST and DAST scan results - are teams keeping up with fixes?
Container security scan results - are teams keeping up with fixes?
Deployment frequency
Unit test coverage
Release speed (from pull request creation to the change being in production)
We also have several "team health" metrics that are in other tools; I'm not sure if this discussion covers those or not.

I'm really interested to hear the latest on what other groups are measuring as well. Have they changed their metrics over the past two years? For us, the importance of the security scans has continued to increase over the past two years.

SAST = Static Application Security Testing. Tools that analyze source code for security issues.

DAST = Dynamic Application Security Testing. Tools that target any service (web apps, web pages, APIs) using HTTP and test for security vulnerabilities and mis-configurations.

SAST and DAST tools create bug reports, and we treat those bug reports like defects. We pull data from these tools (number of bugs, severity, age) via APIs and pull them into our dashboards to calculate security scores.

**Comment by johnmark** [comment]
thanks for kicking this off Tracy. Here's Google Cloud's devops site:
https://cloud.google.com/devops

And they've published a case study from Capital One:
https://cloud.google.com/blog/products/devops-sre/using-the-four-keys-to-measure-your-devops-performance

And here's a project labeled the Four Keys:
https://github.com/GoogleCloudPlatform/fourkeys

**Comment by amfred** [comment]
For softer "team happiness" metrics, here's one Devops-y one that we usually run every quarter: https://engineering.atspotify.com/2014/9/16/squad-health-check-model/

At the corporate level, we also have quarterly employee engagement surveys that ask things like "how likely are you to recommend this as a place to work" and "my manager gives me the support I need".

**Comment by amfred** [comment]
A couple of companies have managed to graph revenue per minute.  You could see the impact of an outage in real time.  No pressure!  It's cool, though.

**Reply by devops-novice** [reply]
A couple of thoughts around metrics and the challenges ahead.

- How to have metrics that capture the happiness quotient of the customer in regard to the DevOps Investments being made by the organizations. 
- How to have metrics that capture the business value that is generated again from the customer perspective to the DevOps Investments being made by the organizations/service provider.

In organizations there are many streams of CoE runs like DevOps, Agile, Ways of Working, Lean, etc... Each has its own stream of metrics. How from the business perspective all those metrics can merge else there will be conflicting situations of whom to please and follow.  
How can DevOps play a bigger brother role in accommodating all / most of them?
